lucet-wasi-sdk/tests/lucetc.rs

```diff
@@ -117,4 +128,39 @@ mod lucetc_tests {
         */
         let _obj = c.object_file().expect("generate code from c & d");
     }
+
+    #[test]
+    fn hello() {
+        let m = {
+            // This is like module_from_c, except missing the bits that have it leave out the wasi
+            // stdlib and tentry points:
+            let tempdir = tempfile::Builder::new()
+                .prefix("wasi-sdk-test")
+                .tempdir()
+                .expect("create tempdir");
+            let mut wasm = PathBuf::from(tempdir.path());
+            wasm.push("out.wasm");
+
+            let linker = Link::new(&[PathBuf::from("tests/hello.c")]);
+            linker.link(wasm.clone()).expect("link");
+
+            let mut wasm_file = File::open(wasm).expect("open wasm");
+            let mut wasm_contents = Vec::new();
+            wasm_file
+                .read_to_end(&mut wasm_contents)
+                .expect("read wasm");
+            wasm_contents
+        };
+
+        let b =
+            Bindings::from_file("../lucet-wasi/bindings.json").expect("load lucet-wasi bindings");
+        let h = HeapSettings::default();
+        let v = Validator::load("../wasi/phases/unstable/witx/wasi_unstable_preview0.witx")
+            .expect("wasi spec validation")
+            .with_wasi_exe(true);
+        let c =
+            Compiler::new(&m, OptLevel::default(), &b, h, false, &Some(v)).expect("compile empty");
+        let mdata = c.module_data().unwrap();
+        assert!(mdata.heap_spec().is_some());
```

Comment:
Could you add some comments that describe what this is testing for at a higher level?